schemas/pulse-job.yml

```diff
@@ -204,28 +241,180 @@ properties:
   runMachine:
     $ref: "#/definitions/machine"
 
-  artifacts:
-    type: "array"
+  textLogSummary:
+    type: object
+    description: |
+      This object defines what is seen in the Treeherder Log Viewer.  This object
+      can be submitted here, or it will be generated from the submitted log.
+    properties:
+      step_data:
```

Comment:
too many nested objects, that don't have to be here... You can merge all properties of `step_data` into `textLogSummary`.